sql/hive/src/test/scala/org/apache/spark/sql/hive/test/TestHive.scala

```diff
@@ -588,20 +588,29 @@ private[hive] class TestHiveQueryExecution(
 
   override lazy val analyzed: LogicalPlan = {
     val describedTables = logical match {
-      case CacheTableCommand(tbl, _, _, _) => tbl.table :: Nil
+      case CacheTableCommand(tbl, _, _, _) => (tbl.table, tbl.database) :: Nil
       case _ => Nil
     }
 
     // Make sure any test tables referenced are loaded.
     val referencedTables =
       describedTables ++
-        logical.collect { case UnresolvedRelation(ident) => ident.last }
+        logical.collect { case UnresolvedRelation(ident) => (ident.last, ident.init.lastOption) }
     val resolver = sparkSession.sessionState.conf.resolver
-    val referencedTestTables = sparkSession.testTables.keys.filter { testTable =>
-      referencedTables.exists(resolver(_, testTable))
+    val referencedTestTables = referencedTables.flatMap { case (table, dbOpt) =>
+      val testTableOpt = sparkSession.testTables.keys.find(resolver(_, table))
+      testTableOpt.map(testTable => Seq((testTable, dbOpt))).getOrElse(Seq.empty)
```

Comment:
nit: I think we can just do `testTableOpt.map(testTable => (testTable, tbl.database))`